Makefile.js

```diff
@@ -790,7 +791,9 @@ target.checkRuleFiles = function() {
         const basename = path.basename(filename, ".js");
         const docFilename = `docs/rules/${basename}.md`;
         const docText = cat(docFilename);
+        const docMarkdown = marked.lexer(docText, { gfm: true, silent: false });
         const ruleCode = cat(filename);
+        const knownHeaders = ["Rule Details", "Options", "Environments", "Examples", "Known Limitations", "When Not To Use It", "Related Rules", "Compatibility", "Further Reading"];
```

Comment:
It wasn't in that order in `guard-for-in.md`, but that was indeed the only one. The order of "Related Rules" and "Compatibility" has more inconsistencies.

My reasoning for the current order of `"Related Rules", "Compatibility", "Further Reading"` in the PR was "Internal rule links", "External rule links" and "Other external links", so progressing from internal links to external links.

But as I said in my initial PR it's a bit of an arbitrary choice, so I'm willing to change it to `"Further Reading", "Related Rules", "Compatibility"` or even something else if that's the consensus.

Reply:
Further Reading is meant to always be last because it’s intended to link outside of the ESLint site. We don’t want to send people off site in the middle of the page. 

I like the order of Related Rules, Compatiblity, then Further Reading.

Reply:
`"Further Reading"` is usually more related to the rule than the related rules are. It provides more info on the context of the problem the rule aims to solve, while `"Related Rules"` is more like leaving the rule to read about another rule. I think that's the reason why the convention was to have `"Further Reading"` before `"Related Rules"`, even though they are external links.

"Further Reading", "Related Rules", "Compatibility" makes sense to me. @nzakas what do you think about this?

Reply:
> Further Reading is meant to always be last because it’s intended to link outside of the ESLint site. We don’t want to send people off site in the middle of the page.
> 
> I like the order of Related Rules, Compatiblity, then Further Reading.

For some reason, I didn't see this post while I was writing my previous one. Okay, if that was the intent then we should follow it.